tests/extension_cases/functions_json.rs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Tests for datafusion-function-json integration
+
+use crate::TestExecution;
+
+static TEST_TABLE: &str = r#"
+CREATE TABLE test_table (
+    id INT,
+    json_col VARCHAR
+) AS VALUES
+(1, '{}'),
+(2, '{ "a": 1 }'),
+(3, '{ "a": 2 }'),
+(4, '{ "a": 1, "b": 2 }'),
+(5, '{ "a": 1, "b": 2, "c": 3 }')
+"#;
+
+/// Ensure one of the functions `json_contains` function is properly registered
+#[tokio::test]
+async fn test_basic() {
+    let mut execution = TestExecution::new().with_setup(TEST_TABLE).await;
+
+    let actual = execution
+        .run_and_format("SELECT id, json_contains(json_col, 'b') as json_contains FROM test_table")
+        .await;
+
+    insta::assert_yaml_snapshot!(actual, @r###"
+    - +----+---------------+
+    - "| id | json_contains |"
+    - +----+---------------+
+    - "| 1  | false         |"
+    - "| 2  | false         |"
+    - "| 3  | false         |"
+    - "| 4  | true          |"
+    - "| 5  | true          |"
+    - +----+---------------+
+    "###);
+}
```

Comment:
I have not used `insta` before so i find this a little confusing (in particular its not clear to me what the "yaml" is doing). But I have heard good things about `insta` so will start looking into it.

Reply:
My TLDR is:
1. Just follow the model (I don't really know what `assert_yaml_snapshot` does internally
2. When the test fails you run `cargo insta` which will a) show you a diff and b) offer to update the expected text for you.

b is the game changer -- you can update a bunch of tests simply by running `cargo insta` and typing `a` after looking at the diffs

Reply:
That sounds pretty cool!

Reply:
It is like `sqllogictest --complete` mode but for rust unit tests 🚀